src/test/java/com/karankumar/bookproject/backend/entity/UserTest.java

```diff
@@ -115,4 +122,16 @@ void testPasswordWithNoSpecialCharacterIsInvalid() {
         // then
         Assertions.assertEquals(initialNumberOfUsers, userRepository.count());
     }
+
+    @Test
+    void testUserWithInvalidMail() {
+        // given
+        User user = userWithInvalidMail().build();
```

Comment:
Could you delete the `userWithInvalidMail()` method and instead create a new method called `createUserWithNoEmail()` (or something similar)?

Then, you can use that method and specify the email in this method.

Reply:
As alluded to before, it's great that we're testing that email addresses without a domain are not accepted. However, we have not yet fully tested the regex. For example, other tests could include email addresses that do not include the `@` sign, email addresses without a top-level domain, email addresses with nothing before the `@` sign and so on. 

In short, we want to fully test all of the ways that someone may try to get around the regex (rather than simply looking at the regex and figuring out the edge cases). This way, we can have a higher level of surety that our email regex works as desired.

Reply:
Ok, I will extract the regex as a constant in  a utility class and I will add some tests (in a new test file) to improve our confidence level about the regex. Do you think is it ok?

Thanks,
Diego

Reply:
I think it's best to keep the regex where it is for now and to not create a new test file for the time being.

Reply:
Ok, thanks. I will work on it in the next days.

Reply:
I added tests for some invalid cases but they will never be exhaustive. Let me know if some other case comes up to your mind.

Thanks,
Diego

Reply:
The additions are good. Thanks a lot!